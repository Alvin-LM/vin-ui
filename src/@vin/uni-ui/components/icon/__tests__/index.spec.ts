import { config, mount } from '@vue/test-utils';
import Icon from '../index.vue';

test('should render icon fabulous', async () => {
  const wrapper = mount(Icon, {
    props: {
      name: 'fabulous',
    },
  });

  expect(wrapper.html()).toMatchSnapshot();
});
test('should render icon image type', async () => {
  const wrapper = mount(Icon, {
    props: {
      name: 'https://vingogo.cn/docs/images/logo.png',
    },
  });

  expect(wrapper.html()).toMatchSnapshot();
});

test('should render icon fabulous color', async () => {
  const wrapper = mount(Icon, {
    props: {
      name: 'fabulous',
      color: 'red',
    },
  });

  expect(wrapper.find<HTMLElement>('.vin-icon-fabulous').element.style.color).toEqual('red');
  expect(wrapper.html()).toMatchSnapshot();
});

test('should render icon fabulous size', async () => {
  const wrapper = mount(Icon, {
    props: {
      name: 'fabulous',
      size: '24',
    },
  });

  expect(wrapper.find<HTMLElement>('.vin-icon-fabulous').element.style.fontSize).toEqual('24px');
  expect(wrapper.html()).toMatchSnapshot();
});
