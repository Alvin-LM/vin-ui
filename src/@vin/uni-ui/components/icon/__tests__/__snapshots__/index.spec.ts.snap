// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render icon fabulous 1`] = `"<text class=\\"vin-iconfont vin-icon vin-icon-fabulous\\"></text>"`;

exports[`should render icon fabulous color 1`] = `"<text style=\\"color: red;\\" class=\\"vin-iconfont vin-icon vin-icon-fabulous\\"></text>"`;

exports[`should render icon fabulous size 1`] = `"<text style=\\"font-size: 24px; width: 24px; height: 24px;\\" class=\\"vin-iconfont vin-icon vin-icon-fabulous\\"></text>"`;

exports[`should render icon image type 1`] = `"<img src=\\"https://cdn.vingogo.cn/logo.png\\" class=\\"vin-icon__img\\">"`;
